app/src/main/java/org/mozilla/fenix/home/tabs/TabsUIView.kt

```diff
@@ -50,8 +51,11 @@ class TabsUIView(
         }
         header.add_tab_button.increaseTapArea(HomeFragment.addTabButtonIncreaseDps)
         header.add_tab_button.setOnClickListener {
-            val directions = HomeFragmentDirections.actionHomeFragmentToSearchFragment(null)
-            Navigation.findNavController(it).navigate(directions)
+            if (view.context as? HomeActivity != null) {
```

Comment:
This is deep inside the Component. It seems limiting to require that the Tabs UI Component only exists inside of a HomeActivity. One could pass isPrivate into the ViewState with the initialState and possibly with Changes.

I know it seems unlikely we'll place another tab switcher somewhere else in the UI or even that we'll have it in another activity, but it's also really messy to reference the Activity at this level. We don't know if our Fragment is even attached.

Let me know if I can help here with architecture improvements as I think the initialState should be more available to the UIView than it is.